src/pages/settings/Profile/PersonalDetails/LegalNamePage.js

```diff
@@ -36,92 +37,72 @@ const defaultProps = {
     },
 };
 
-class LegalNamePage extends Component {
-    constructor(props) {
-        super(props);
+function LegalNamePage(props) {
+    const legalFirstName = lodashGet(props.privatePersonalDetails, 'legalFirstName', '');
+    const legalLastName = lodashGet(props.privatePersonalDetails, 'legalLastName', '');
 
-        this.validate = this.validate.bind(this);
-        this.updateLegalName = this.updateLegalName.bind(this);
-    }
-
-    /**
-     * Submit form to update user's legal first and last name
-     * @param {Object} values
-     * @param {String} values.legalFirstName
-     * @param {String} values.legalLastName
-     */
-    updateLegalName(values) {
+    const updateLegalName = (values) => {
```

Comment:
We landed on some standard where we will [move the function outside of the component if it doesn't need to be recreated and otherwise just always do the `useCallback()` thing](https://expensify.slack.com/archives/C01GTK53T8Q/p1680096510744619). 

But maybe you have something to add to that conversation though @arosiclair (I have no strong feelings on it and mostly just wanted to settle on some "best practice").

Reply:
Moving component functions out of the component feels icky, but I can't really argue with that conclusion so I'll update this to follow

Reply:
definitely up for discussing and changing it if you have any more thoughts. Something about it does feel off to me - but I get the idea of defaulting to something with a stable reference (even if you aren't gonna need it).

Reply:
Yeah I guess until I have a better argument than "it feels icky" I'll just follow that discussion 😄